packages/flutter_tools/lib/src/commands/drive.dart

```diff
@@ -116,30 +115,29 @@ class DriveCommand extends RunCommandBase {
         );
       }
 
-      int result = await appStarter(this);
-      if (result != 0)
-        throwToolExit('Application failed to start ($result). Will not run test. Quitting.', exitCode: result);
+      LaunchResult result = await appStarter(this);
+      if (result == null)
+        throwToolExit('Application failed to start. Will not run test. Quitting.', exitCode: 1);
+      if (!result.started)
+        throwToolExit('Application failed to start. Will not run test. Quitting.', exitCode: 2);
```

Comment:
I just removed the case where it returns 2, and always returned 1. Turns out the tests wanted that anyway.